code/modules/mob/living/autohiss.dm

```diff
@@ -82,8 +82,44 @@
 		)
 	autohiss_exempt = list(LANGUAGE_VAURCA)
 
+/datum/species/diona
+	autohiss_basic_map = list(
+			"s" = list("ss","sss"),
+			"z" = list("zz","zzz"),
+			"e" = list("ee", "eee")
+		)
+	autohiss_extra_map = list(
+			"a" = list("aa", "aaa"),
+			"i" = list("ii", "iii"),
+			"o" = list("oo", "ooo"),
+			"u" = list("uu", "uuu")
+		)
+	autohiss_exempt = list(
+			LANGUAGE_ROOTSONG
+		)
+
+/datum/species/proc/handle_autoslow(message)
+
+	var/returning = ""
+	var/longwords = list(
+		"who","what","when","where","why","how",
+		"i'm","i","am","this",
+		"they","are","they're","their","his","her","their","the","he","she",
+		)
+
+	for(var/word in text2list(message," ")) // For each word in a message
+		var/addum = word + " "
+		if (lowertext(word) in longwords)
+			addum = word + "... "
+		returning += addum
+
+	return trim(returning)
 
 /datum/species/proc/handle_autohiss(message, datum/language/lang, mode)
+
+	if ( (mode && mode > 0) && (name && name == "Diona") && (lang && lang.name && lang.name != LANGUAGE_ROOTSONG) )

```

Comment:
This can be simplified a bunch, and should be made more generic so other species can implement custom autohiss as well.

Equivalent:
```dm
if (mode > 0 && name == "Diona" && lang && lang.name != LANGUAGE_ROOTSONG)
```

Reply:
Is this even necessary? Override diona autohiss with autoslow and voila.

```dm
/datum/species/diona/handle_autohiss(message, datum/language/lang, mode)
	if (lang && lang.name == LANGUAGE_ROOTSONG)
		// autoslow in here, make sure it modifies message

	. = ..(message, lang, mode)
```

Reply:
Ye, I suggested this in another comment.